src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1501,40 +1692,72 @@ public DatabaseTopology ReadDatabaseTopology(BlittableJsonReaderObject rawDataba
             return JsonDeserializationCluster.DatabaseTopology(topology);
         }
 
-        public IEnumerable<(string Prefix, long Value)> ReadIdentities<T>(TransactionOperationContext<T> context, string databaseName, int start, long take)
-            where T : RavenTransaction
+        public IEnumerable<(string Prefix, long Value, long index)> GetIdentitiesFrom(TransactionOperationContext context, string dbName, long fromIndex, long take)
         {
-            var identities = context.Transaction.InnerTransaction.ReadTree(Identities);
+            CompareExchangeCommandBase.GetPrefixIndexSlices(context.Allocator, dbName, fromIndex, out var indexTuple);
```

Comment:
Shouldn't we have a `using` here?

Reply:
GetPrefixIndexSlices is void, I do use using afterwards on the out indexTuple

Reply:
added